core/src/main/scala/scalaz/StrictTree.scala

```diff
@@ -0,0 +1,208 @@
+    def point[A](a: => A): StrictTree[A] = StrictTree.Leaf(a)
+    def cobind[A, B](fa: StrictTree[A])(f: StrictTree[A] => B): StrictTree[B] = fa cobind f
+    def copoint[A](p: StrictTree[A]): A = p.rootLabel
+    override def map[A, B](fa: StrictTree[A])(f: A => B) = fa map f
+    def bind[A, B](fa: StrictTree[A])(f: A => StrictTree[B]): StrictTree[B] = fa flatMap f
+    def traverse1Impl[G[_]: Apply, A, B](fa: StrictTree[A])(f: A => G[B]): G[StrictTree[B]] = fa traverse1 f
+    override def foldRight[A, B](fa: StrictTree[A], z: => B)(f: (A, => B) => B): B = fa.foldRight(z)(f)
+    override def foldMapRight1[A, B](fa: StrictTree[A])(z: A => B)(f: (A, => B) => B) = (fa.flatten.reverse: @unchecked) match {
+      case h +: t => t.foldLeft(z(h))((b, a) => f(a, b))
+    }
+    override def foldLeft[A, B](fa: StrictTree[A], z: B)(f: (B, A) => B): B =
+      fa.flatten.foldLeft(z)(f)
+    override def foldMapLeft1[A, B](fa: StrictTree[A])(z: A => B)(f: (B, A) => B): B = fa.flatten match {
+      case h +: t => t.foldLeft(z(h))(f)
+    }
+    override def foldMap[A, B](fa: StrictTree[A])(f: A => B)(implicit F: Monoid[B]): B = fa foldMap f
+    def zip[A, B](aa: => StrictTree[A], bb: => StrictTree[B]): StrictTree[(A, B)] = {
+      val a = aa
+      val b = bb
+      StrictTree.Node(
+        (a.rootLabel, b.rootLabel),
+        Zip[Vector].zipWith(a.subForest, b.subForest)(zip(_, _))
+      )
+    }
+    def alignWith[A, B, C](f: (\&/[A, B]) ⇒ C): (StrictTree[A], StrictTree[B]) => StrictTree[C] = {
+      def align(ta: StrictTree[A], tb: StrictTree[B]): StrictTree[C] =
+        StrictTree.Node(f(\&/(ta.rootLabel, tb.rootLabel)), Align[Vector].alignWith[StrictTree[A], StrictTree[B], StrictTree[C]]({
+          case \&/.This(sta) ⇒ sta map {a ⇒ f(\&/.This(a))}
+          case \&/.That(stb) ⇒ stb map {b ⇒ f(\&/.That(b))}
+          case \&/(sta, stb) ⇒ align(sta, stb)
+        })(ta.subForest, tb.subForest))
+      align _
+    }
+    def zip[A, B](a: StrictTree[A], b: StrictTree[B]): StrictTree[(A, B)] = {
+      StrictTree.Node(
+        (a.rootLabel, b.rootLabel),
+        Zip[Vector].zipWith(a.subForest, b.subForest)(zip(_, _))
+      )
+    }
+  }
+
+  implicit def treeEqual[A](implicit A0: Equal[A]): Equal[StrictTree[A]] =
+    new StrictTreeEqual[A] { def A = A0 }
+
+  implicit def treeOrder[A](implicit A0: Order[A]): Order[StrictTree[A]] =
+    new Order[StrictTree[A]] with StrictTreeEqual[A] {
+      def A = A0
+      import std.vector._
+      override def order(x: StrictTree[A], y: StrictTree[A]) =
+        A.order(x.rootLabel, y.rootLabel) match {
+          case Ordering.EQ =>
+            Order[Vector[StrictTree[A]]].order(x.subForest, y.subForest)
+          case x => x
+        }
+    }
+
+  /* TODO
+  def applic[A, B](f: StrictTree[A => B]) = a => StrictTree.node((f.rootLabel)(a.rootLabel), implicitly[Applic[newtypes.ZipVector]].applic(f.subForest.map(applic[A, B](_)).?)(a.subForest ?).value)
+   */
```

Comment:
@aloiscochard It's a TODO for Tree, so I copied it to StrictTree. You'd have to ask @retronym what the status is.